src/gui/graphview.cpp

```diff
@@ -73,6 +73,27 @@ GraphView::Star GraphView::createStar(const Node& node,
     return star;
 }
 
+void GraphView::slotUpdateSelection() {
+    for (auto selectedNode : m_selectedNodes) {
+        int i = selectedNode.first;
+        const Node node = selectedNode.second;
+        const QPointF xy = QPointF(node.x() * currEdgeSize(), node.y() * currEdgeSize());
+
+        Star selectedStar(node, xy, {});
+        selectedStar.edges.reserve(node.outEdges().size());
+        for (auto const& ep : node.outEdges()) {
+            QPointF xy2 = nodePoint(ep.second.neighbour(), currEdgeSize());
+            QLineF line(xy, xy2);
+            if (!m_showNodes || line.length() - m_nodeRadius * 2. > 4.0) {
+                selectedStar.edges.push_back({ ep.second, line });
+            }
+        }
+        selectedStar.edges.shrink_to_fit();
```

Comment:
this approach isn't very efficient either
because you'll always end up looping through all nodes twice

actually, this code is mostly a copy&paste from `createStar()`. Could we just call the `createStart()` function straight away when it's needed? (i.e., instead of emitting the nodesMoved() signals)

Reply:
> Could we just call the ``createStart()`` function straight away when it's needed? (i.e., instead of emitting the ``nodesMoved()`` signals)

Hm not sure I quite get that. Do you mean to refactor this without using signals?

Reply:
yes, there's no gain in using the signal&slot in this case

Reply:
My main point here is that your "slotUpdateSelection" is just a copy&paste from "createStar"; we don't need to duplicate this code.

Reply:
I reformatted a bit as to directly call ``createStar``, but I still can't see a more efficient way to call the ``UpdateSelection``. This can't be called directly from the painting methods and neither can it not be called each time a node is moved. What am I missing?